src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -327,6 +328,9 @@ public IList<string> Merge (List<TypeDefinition> subclasses, List<string> select
 
 			var providerNames = AddMonoRuntimeProviders (app);
 
+			if (MultiDex)
+				app.Add (CreateMonoRuntimeProvider ("mono.android.MultiDexLoader", null, initOrder: int.MaxValue));
```

Comment:
I think this should be `AppInitOrder+1`, not `int.MaxValue`, just in case someone else needs to go higher still.